src/core/include/openvino/core/extension.hpp

```diff
@@ -28,24 +28,28 @@ class OPENVINO_API Extension {
 
     virtual ~Extension();
 };
+}  // namespace ov
 
+#ifndef IE_CREATE_EXTENSION
```

Comment:
why do we need this `ifndef`
If it's for compatibility with old tests, then they can be removed

Reply:
@ilya-lavrenov  This can not be removed! This define is used for static library, which needs different func name .
https://github.com/openvinotoolkit/openvino/blob/9638b899db494ab7f8424191b15684f598a9f89a/cmake/developer_package/plugins/plugins.cmake#L75-L85